File: 프로그래머스/1/42889. 실패율/실패율.py
def solution(N, stages):
    answer = [0] * (N + 1)
    arr = [0.0] * (N + 1)
    peop = len(stages)

    for stage in stages:
        if stage <= N:
            answer[stage] += 1

    for i in range(1, N + 1):
        if peop == 0:
            arr[i] = 0
        else:
            arr[i] = answer[i] / peop
            peop -= answer[i]

    index = list(range(1, N + 1))

    index.sort(key=lambda x: (-arr[x], x))

    return index

File: 프로그래머스/1/42889. 실패율/실패율.java
import java.util.*;

class Solution {
    public Integer[] solution(int N, int[] stages) {
        int[] answer = new int[N+1];
        double[] arr = new double[N+1];
        int peop = stages.length;

        for(int i = 0; i<stages.length; i++) {
            if(stages[i] < N+1)
                answer[stages[i]] += 1;
        }

        for(int i = 1; i<N+1; i++){
            arr[i] = answer[i]==0?0:(double)answer[i]/peop;
            peop -= answer[i];
        }

        Integer[] index = new Integer[N];
        for (int i = 0; i < N; i++) {
            index[i] = i+1;
        }

        Arrays.sort(index, (a, b) -> {
            if (arr[b] != arr[a]) {
                return Double.compare(arr[b], arr[a]);
            } else {
                return Integer.compare(a, b);
            }
        });
        return index;
    }
}